src/Robin/Ntlm/Crypt/Hasher/AbstractHasher.php

```diff
@@ -90,14 +94,14 @@ public function digest()
      *
      * @link http://php.net/manual/en/hash.resources.php
      * @param mixed $context The context to validate.
-     * @return resource The incremental hashing context.
+     * @return resource|HashContext The incremental hashing context.
      * @throws InvalidArgumentException If the hash context isn't valid.
      */
     protected function validateHashContext($context)
     {
-        if (false === $context
+        if (!($context instanceof HashContext) && (false === $context
```

Comment:
Is the `HashContext` class 100% interchangeable with the `Hash Context` resource type?

Reply:
It _should_ be (I believe), as its just used as an argument to functions to keep an internal context and has no methods.

https://www.php.net/manual/en/class.hashcontext.php
https://www.php.net/manual/en/function.hash-init.php#refsect1-function.hash-init-changelog